Assets/MixedRealityToolkit/InputSystem/Sources/GestureInputSource.cs

```diff
@@ -0,0 +1,191 @@
+            }
+            else
+            {
+                navigationGestureRecognizer.SetRecognizableGestures(GestureSettings.NavigationX | GestureSettings.NavigationY | GestureSettings.NavigationZ);
+            }
+
+            if (gestureRecognizer != null && !gestureRecognizer.IsCapturingGestures())
+            {
+                gestureRecognizer.StartCapturingGestures();
+            }
+
+            if (navigationGestureRecognizer != null && !navigationGestureRecognizer.IsCapturingGestures())
+            {
+                navigationGestureRecognizer.StartCapturingGestures();
+            }
+        }
+
+        /// <inheritdoc />
+        public override void Dispose()
+        {
+            if (gestureRecognizer != null && gestureRecognizer.IsCapturingGestures())
+            {
+                gestureRecognizer.StopCapturingGestures();
+            }
+
+            if (navigationGestureRecognizer != null && navigationGestureRecognizer.IsCapturingGestures())
+            {
+                navigationGestureRecognizer.StopCapturingGestures();
+            }
+
+            if (gestureRecognizer != null)
+            {
+                gestureRecognizer.Tapped -= GestureRecognizer_Tapped;
+
+                gestureRecognizer.HoldStarted -= GestureRecognizer_HoldStarted;
+                gestureRecognizer.HoldCompleted -= GestureRecognizer_HoldCompleted;
+                gestureRecognizer.HoldCanceled -= GestureRecognizer_HoldCanceled;
+
+                gestureRecognizer.ManipulationStarted -= GestureRecognizer_ManipulationStarted;
+                gestureRecognizer.ManipulationUpdated -= GestureRecognizer_ManipulationUpdated;
+                gestureRecognizer.ManipulationCompleted -= GestureRecognizer_ManipulationCompleted;
+                gestureRecognizer.ManipulationCanceled -= GestureRecognizer_ManipulationCanceled;
+
+                gestureRecognizer.Dispose();
+            }
+
+            if (navigationGestureRecognizer != null)
+            {
+                navigationGestureRecognizer.NavigationStarted -= NavigationGestureRecognizer_NavigationStarted;
+                navigationGestureRecognizer.NavigationUpdated -= NavigationGestureRecognizer_NavigationUpdated;
+                navigationGestureRecognizer.NavigationCompleted -= NavigationGestureRecognizer_NavigationCompleted;
+                navigationGestureRecognizer.NavigationCanceled -= NavigationGestureRecognizer_NavigationCanceled;
+
+                navigationGestureRecognizer.Dispose();
+            }
+        }
+
+        private readonly GestureRecognizer gestureRecognizer;
+        private readonly GestureRecognizer navigationGestureRecognizer;
+
+        #region Raise GestureRecognizer Events
+
+        private void GestureRecognizer_Tapped(TappedEventArgs args)
+        {
+            InputSystem.RaisePointerClicked(InputSystem.GazeProvider.GazePointer, (Handedness)args.source.handedness, PointerAction, args.tapCount);
```

Comment:
Because it's a pointer event not a key press.

Reply:
It's a trigger click, if you're using a controller.

Reply:
We'll be sure to filter the controllers out here, then.

Reply:
Why the down thumbs? 😅 I thought that's what you were suggesting.

Reply:
Anything that is handled by the controller should happen in the controller implementation, not in this class, so it would be valid to remove anything not related to gesture control.

Reply:
>I thought that's what you were suggesting.

Nope! I think everything that comes through this recognizer is valid and should be turned into the appropriate events, without filtering by type. 

I don't think the gesture recognizer being its own input source makes sense. The events are coming from the same input source as the source detected / input down / etc events from the controller implementation, so it makes sense to me that these should reference the same Controller/source.

I also think pointer clicked for the UWP WMR controller implementation makes more sense to happen here, since it's a better defined "click" event than using an input up event, and provides things like `tapCount`.

I started an implementation of the gesture recognizer a week or so ago that maintains the references to the input source and their pointers and whatnot. What do we gain by making this separate from the controller implementation?

Reply:
I figured out a way to get the controller references from the windows device manager. Not sure how @SimonDarksideJ will like this, but I thought it was an interesting fix to the issue without filtering out the controllers

Reply:
> I don't think the gesture recognizer being its own input source makes sense. The events are coming from the same input source as the source detected / input down / etc events from the controller implementation, so it makes sense to me that these should reference the same Controller/source.

I do agree with you here, but @SimonDarksideJ made it very clear he wanted to separate these out.

> I started an implementation of the gesture recognizer a week or so ago

Well where's that at, and why is it the first time we're hearing about it?